### PR TITLE
docs(AttachmentBuilder): fix #8407

### LINKED ISSUE #8407: AttachmentBuilder's `data` parameter may be of the wrong documented type
### Which package is this bug report for?

discord.js

### Issue description

The [`AttachmentBuilder`](https://discord.js.org/#/docs/discord.js/main/class/AttachmentBuilder)'s second parameter in the constructor is an [`APIAttachment`](https://discord.com/developers/docs/resources/channel#attachment-object) which seems to document 10 properties currently. However, as seen here:

https://github.com/discordjs/discord.js/blob/4d8361c711df423f154a7460939c60f6d9429105/packages/discord.js/src/structures/AttachmentBuilder.js#L9-L29

Only 1 seems to be used which is `description` (and `filename` seems to be `name` here). This doesn't look correct.

In the typings, however:

https://github.com/discordjs/discord.js/blob/4d8361c711df423f154a7460939c60f6d9429105/packages/discord.js/typings/index.d.ts#L1745

https://github.com/discordjs/discord.js/blob/4d8361c711df423f154a7460939c60f6d9429105/packages/discord.js/typings/index.d.ts#L3730-L3733

This seems to be typed correctly. The documentation is very likely outdated.

### Code sample

_No response_

### Package version

14.1.3-dev

### Node.js version

Node.js 16.14.0; TypeScript 4.7.4

### Operating system

macOS Monterey 12.4

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

Not applicable (subpackage bug)

### Which gateway intents are you subscribing to?

Not applicable (subpackage bug)

### I have tested this issue on a development release

4d8361c

### PR DESCRIPTION
**Pull request by RedGuy12**
**Please describe the changes this PR makes and why it should be merged:**

Changes the type in the JSDoc for AttachmentBuilder to fix #8407

**Status and versioning classification:**

- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/discord.js/src/structures/AttachmentBuilder.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const { basename, flatten } = require('../util/Util');
 class AttachmentBuilder {
   /**
    * @param {BufferResolvable|Stream} attachment The file
-   * @param {APIAttachment} [data] Extra data
+   * @param {AttachmentData} [data] Extra data
```
Comment (Jiralite) [inline packages/discord.js/src/structures/AttachmentBuilder.js:11]: It seems this type definition doesn't exist. Mind creating it?

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/AttachmentBuilder.js (hunk 1 above)
Comment (RedGuy12) [reply packages/discord.js/src/structures/AttachmentBuilder.js:11]: It exists...https://github.com/discordjs/discord.js/blob/8683562704dcde1fa3d58dc6a1e774c267228f84/packages/discord.js/typings/index.d.ts#L3730

Comment (Jiralite) [reply packages/discord.js/src/structures/AttachmentBuilder.js:11]: It doesn't exist in JSDoc.

Comment (RedGuy12) [reply packages/discord.js/src/structures/AttachmentBuilder.js:11]: oh, I assumed that the types in the TS typedef were automatically synced, I see now that's not the case. Fixed in ae86ea1607948d54b07d4ccdca80951e2be40a0c.

File: packages/discord.js/src/structures/AttachmentBuilder.js [hunk 2]
```diff
@@ -108,3 +108,9 @@ module.exports = AttachmentBuilder;
  * @external APIAttachment
  * @see {@link https://discord.com/developers/docs/resources/channel#attachment-object}
  */
+
+/**
+ * @typedef {Object} AttachmentData
+ * @property {name} [string] The name of the attachment
+ * @property {description} [string] The description of the attachment
```
Comment (Jiralite) [inline packages/discord.js/src/structures/AttachmentBuilder.js:115]:
```suggestion
 * @property {string} [name] The name of the attachment
 * @property {string} [description] The description of the attachment
```

Comment (RedGuy12) [reply packages/discord.js/src/structures/AttachmentBuilder.js]: 🤦🏻 Whoops, I was a bit in a rush when I wrote that 😅

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**